src/mlpack/methods/ann/layer/residual.hpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * @file methods/ann/layer/residual.hpp
+ * @author Shubham Agrawal
+ *
+ * Base class for neural network layers that are wrappers around other layers.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RESIDUAL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RESIDUAL_HPP
+
+#include "../make_alias.hpp"
+#include "multi_layer.hpp"
+
+namespace mlpack {
+namespace ann {
+
+/**
+ * A lambda "map-reduce" is a layer that is a wrapper around other layers.  
```

Comment:
I think this comment needs to be updated.  Also it would be a good idea to be clear about precisely how the output is reduced (i.e. along which axis, etc.) so that a user can reason about what the output dimensions of one of these layers is.